pkg/controllers/componentroutesecretsync/component_route_secret_sync_controller.go

```diff
@@ -0,0 +1,86 @@
+package componentroutesecretsync
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+)
+
+func NewComponentRouteSecretSyncCrontroller(
+	componentRouteNamespace string,
+	componentRouteName string,
+	destSecretName string,
+	destSecretNamespace string,
+	configInformer configinformers.IngressInformer,
+	secretInformer corev1informers.SecretInformer,
+	eventRecorder events.Recorder,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+) factory.Controller {
+	controller := &componentRouteSecretSyncController{
+		destSecretNamespace: destSecretNamespace,
+		destSecretName:      destSecretName,
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		ingressLister:  configInformer.Lister(),
+		secretLister:   secretInformer.Lister(),
+		resourceSyncer: resourceSyncer,
+	}
+	return factory.New().
+		WithInformers(
+			configInformer.Informer(),
+			secretInformer.Informer(),
+		).
+		WithSync(controller.sync).
+		ResyncEvery(1*time.Minute).
+		ToController("componentRouteSecretSyncController", eventRecorder)
+}
+
+type componentRouteSecretSyncController struct {
+	destSecretNamespace string
+	destSecretName      string
+	componentRoute      types.NamespacedName
+	ingressLister       configlistersv1.IngressLister
+	secretLister        corev1listers.SecretLister
+	resourceSyncer      resourcesynccontroller.ResourceSyncer
+}
+
+func (c *componentRouteSecretSyncController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
```

Comment:
also add componentRoute status reporting

Reply:
I was thinking of adding that in a second controller, but I can do that here.